lib/concordance_validation/delta.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require "zlib"
+require "settings"
+
+module ConcordanceValidation
+
+  # Performs a diff of two validated concordances
+  class Delta
+    attr_accessor :old_conc, :new_conc, :adds, :deletes
+
+    def initialize(old_conc_filename, new_conc_filename)
+      @old_conc = open_concordance(old_conc_filename)
+      @new_conc = open_concordance(new_conc_filename)
+      @adds = Hash.new {|h, resolved| h[resolved] = Set.new }
+      @deletes = Hash.new {|h, resolved| h[resolved] = Set.new }
+    end
+
+    def run
+      old_conc_lines = old_conc.readlines.to_set
```

Comment:
`comm` is more efficient memory-wise in that it doesn't need to read the entire old file into memory; rather, it relies on both files being sorted and does a sort of merge. Given that this diff format doesn't seem to be totally working for us and that we might need to make some other changes here anyway to reduce churn, I think this is OK for the time being.

Reply:
While it's definitely less efficient than `comm`, `Delta.new.run` only took ~10 minutes so worth the trade off for future flexibility.

Reply:
:+1: